public/angular/js/controllers/users/profile.js

```diff
@@ -0,0 +1,28 @@
+angular.module('Aggie')
+
+.controller('UsersProfileController', [
+  '$scope',
+  '$rootScope',
+  '$stateParams',
+  'users',
+  'User',
+  function($scope, $rootScope, stateParams, users, User) {
+    $scope.users = users;
+
+    function init () {
+      if (!stateParams["userId"] || stateParams["userId"] == "") {
+        $scope.user = $rootScope.currentUser;
+      } else {
+        for (var x = 0; x < $scope.users.length; x++) {
+          if ($scope.users[x]._id === stateParams.userId) {
+            $scope.user = $scope.users[x];
+          }
+        }
```

Comment:
Please use an underscore method or other helper instead of writing this whole thing out.